### PR TITLE
libXL support in Xenmgr

### PR DESCRIPTION
**Pull request by crogers1**
This series of commits provides support for libXL as the lower level toolstack piece, replacing xenvm.  The PR has been broken up into 4 commits and the changes are documented in each commit message.

### CODE REVIEW

**Comment by jean-edouard** [comment]
Fails to build, see http://158.69.127.133:8020/builders/openxt-custom/builds/215

**Comment by jean-edouard** [comment]
The previous comment was a bug in the idl that's now fixed.

File: templates/default/new-vm [hunk 1]
```diff
@@ -4,7 +4,8 @@
     "0": "my-stubdom -> 0:4001",
     "1": "my-stubdom -> 0:4002",
     "2": "my-stubdom -> 0:5000",
-    "3": "my-stubdom -> 0:5001"
+    "3": "my-stubdom -> 0:5001",
+    "4": "my-stubdom -> 0:5100"
```
Comment (jean-edouard) [inline templates/default/new-vm]: Unfortunately this change will need a migration script to handle upgrades...

File: templates/default/service-uivm [hunk 2]
```diff
@@ -66,7 +66,7 @@
     "hvm": "false",
     "apic": "true",
     "nx": "true",
-    "vkbd" : "true",
+    "vkb" : "true",
```
Comment (jean-edouard) [inline templates/default/service-uivm]: Same as above, if we really want to rename this attribute we need a migration script.
Maybe it's best to just keep the old name?

File: xenmgr/Vm/Actions.hs [hunk 3]
```diff
@@ -794,31 +828,65 @@ bootVm config
           liftIO $
             mapM_ updateCdDeviceMediaStatusKey =<< liftIO getHostBSGDevices
 
+      setupBiosStrings uuid =
+          whenDomainID_ uuid $ \domid -> do
+            liftIO $ xsWrite (xsp domid ++ "/bios-strings/xenvendor-manufacturer") "OpenXT"
+            liftIO $ xsWrite (xsp domid ++ "/bios-strings/xenvendor-product") "OpenXT 5.0.0"
```
Comment (jean-edouard) [inline xenmgr/Vm/Actions.hs]: Dunno what that is but the version number seems off...

**Comment by crogers1** [comment]
Addressed the above comments with the latest commit on this branch

**Comment by crogers1** [comment]
Fixed a typo in service-uivm which causes the platform to lose mouse and keyboard input.

File: templates/default/new-vm (hunk 1 above)
Comment (rossphilipson) [inline templates/default/new-vm]: So will the next 3 also it seems. Let's put a ticket in so that doesn't get lost.

**Comment by rossphilipson** [comment]
Merging soon.

**Pull request merged**